src/gui/EditWidgetIcons.cpp

```diff
@@ -196,6 +199,14 @@ void EditWidgetIcons::downloadFavicon()
 #endif
 }
 
+void EditWidgetIcons::showFaviconDialog()
+{
+#ifdef WITH_XC_NETWORKING
+    auto faviconDownloadDialog = new FaviconDownloadDialog(this);
+    faviconDownloadDialog->open();
+#endif
+}
```

Comment:
I would actually prefer an inline text box over a pop up dialog. It reduces code significantly and would be visible to the user at all times.